producer/buffer/buffer.go

```diff
@@ -148,13 +148,14 @@ func (b *buffer) dropEarliestUntilTargetWithLock(targetSize uint64) {
 }
 
 func (b *buffer) Init() {
-	b.closeWG.Add(1)
-	go b.cleanupForever()
+	b.wg.Add(1)
+	go func() {
+		b.cleanupForever()
+		b.wg.Done()
+	}()
 }
 
 func (b *buffer) cleanupForever() {
```

Comment:
super nit: Feel free to ignore but perhaps a more accurate method name would be periodicCleanup() - the cleanupForever to me implies there's no way to end.

Reply:
can rename to `...UntilClose` instead